src/main/java/net/imglib2/algorithm/stats/InformationMetrics.java

```diff
@@ -0,0 +1,185 @@
+		double HB = marginalEntropy( jointHist, 1 );
+		double HAB = entropy( jointHist );
+
+		return HA + HB - HAB;
+	}
+	
+	public static <T extends RealType< T >> HistogramNd<T> jointHistogram(
+			IterableInterval< T > dataA,
+			IterableInterval< T > dataB,
+			double histmin, double histmax, int numBins )
+	{
+		Real1dBinMapper<T> binMapper = new Real1dBinMapper<T>( histmin, histmax, numBins, false );
+		ArrayList<BinMapper1d<T>> binMappers = new ArrayList<BinMapper1d<T>>( 2 );
+		binMappers.add( binMapper );
+		binMappers.add( binMapper );
+
+		List<Iterable<T>> data = new ArrayList<Iterable<T>>( 2 );
+		data.add( dataA );
+		data.add( dataB );
+		return new HistogramNd<T>( data, binMappers );
+	}
+	
+	/**
+	 * Returns the joint entropy of the inputs
+	 * @param rai the RandomAccessibleInterval
+	 * @param ra the RandomAccessible
+	 * @return the joint entropy 
+	 */
+	public static <T extends RealType< T >> double jointEntropy(
+			IterableInterval< T > dataA,
+			IterableInterval< T > dataB,
+			double histmin, double histmax, int numBins )
+	{
+		return entropy( jointHistogram( dataA, dataB, histmin, histmax, numBins ));
+	}
+
+	/**
+	 * Returns the entropy of the input.
+	 * 
+	 * @param data the data
+	 * @return the entropy 
+	 */
+	public static <T extends RealType< T >> double entropy(
+			IterableInterval< T > data,
+			double histmin, double histmax, int numBins )
+	{
+		Real1dBinMapper<T> binMapper = new Real1dBinMapper<T>(
+				histmin, histmax, numBins, false );
+		final Histogram1d<T> hist = new Histogram1d<T>( binMapper );
+		hist.countData( data );
+
+		return entropy( hist );
+	}
+
+	/**
+	 * Computes the entropy of the input 1d histogram.
+	 * @param hist the histogram
+	 * @return the entropy
+	 */ 
+	public static < T > double entropy( Histogram1d< T > hist )
+	{
+		double entropy = 0.0;
+		for( int i = 0; i < hist.getBinCount(); i++ )
+		{
+			double p = hist.relativeFrequency( i, false );
+			if( p > 0 )
+				entropy -= p * Math.log( p );
+
+		}
+		return entropy;
+	}
+
+	/**
+	 * Computes the entropy of the input nd histogram.
+	 * @param hist the histogram
+	 * @return the entropy
+	 */ 
+	public static < T > double entropy( HistogramNd< T > hist )
+	{
+		double entropy = 0.0;
+		Cursor< LongType > hc = hist.cursor();
+		long[] pos = new long[ hc.numDimensions() ];
+
+		while( hc.hasNext() )
+		{
+			hc.fwd();
+			hc.localize( pos );
+			double p = hist.relativeFrequency( pos, false );
+			if( p > 0 )
+				entropy -= p * Math.log( p );
+
+		}
+		return entropy;
+	}
+	
+	public static < T > double marginalEntropy( HistogramNd< T > hist, int dim )
+	{
+		
+		final long ni = hist.dimension( dim );
+		final long total = hist.valueCount();
+		long count = 0;
+		double entropy = 0.0;
+		long ctot = 0;
+		for( int i = 0; i < ni; i++ )
+		{
+			count = subHistCount( hist, dim, i );
+			ctot += count;
+			double p = 1.0 * count / total;
+
+			if( p > 0 )
+				entropy -= p * Math.log( p );
+		}
```

Comment:
Is this something we can run multi-threaded using `LoopBuilder`? See also #83.

Reply:
Good idea, I'll check :+1: thanks @imagejan